src/main/scala/is/hail/expr/Type.scala

```diff
@@ -705,17 +714,34 @@ abstract class TContainer extends Type {
   def elementOffset(region: MemoryBuffer, aoff: Long, i: Int): Long =
     elementOffset(aoff, region.loadInt(aoff), i)
 
+  def elementOffset(aoff: Code[Long], length: Code[Int], i: Code[Int]): Code[Long] =
+    aoff + elementsOffset(length) + i.toL * const(elementByteSize)
+
+  def elementOffsetInRegion(region: Code[MemoryBuffer], aoff: Code[Long], i: Code[Int]): Code[Long] =
```

Comment:
Sign, erasure.  Can you rename the non-staged version for consistency?

Reply:
done.